frontend/src/components/dashboard/DashboardTransactions.tsx

```diff
@@ -1,35 +1,56 @@
 import Link from "next/link";
-
 import React from "react";
+import { Row, Col } from "react-bootstrap";
 
 import TransactionIcon from "../../../public/static/images/icon-t-transactions.svg";
-
-import { Row, Col } from "react-bootstrap";
+import TransactionsList from "../transactions/TransactionsList";
+import TransactionsApi, * as T from "../../libraries/explorer-wamp/transactions";
 
 import Content from "../utils/Content";
-import TransactionsList from "../transactions/TransactionsList";
-import TransactionsApi from "../../libraries/explorer-wamp/transactions";
 import FlipMove from "../utils/FlipMove";
+import PaginationSpinner from "../utils/PaginationSpinner";
 
-export default class extends React.Component {
+interface Props {}
+
+interface State {
+  transactions: T.Transaction[];
+  limit: number;
+}
+
+export default class extends React.Component<Props, State> {
   state = {
-    transactions: this.props.transactions
+    transactions: [],
+    limit: 10
   };
 
+  _transactionsApi: TransactionsApi | null;
+  timer: ReturnType<typeof setTimeout> | null;
+
+  constructor(props: Props) {
+    super(props);
+    this._transactionsApi = null;
+    this.timer = null;
+  }
+
   componentDidMount() {
-    this.regularFetchInfo();
+    this.timer = setTimeout(this.regularFetchInfo, 0);
   }
 
   componentWillUnmount() {
-    clearTimeout(this.timer);
+    clearTimeout(this.timer!);
     this.timer = null;
   }
 
   fetchInfo = async () => {
-    const transactions = await new TransactionsApi()
-      .getLatestTransactionsInfo(10)
-      .catch(() => null);
-    this.setState({ transactions });
+    if (this._transactionsApi === null) {
+      this._transactionsApi = new TransactionsApi();
+    }
+    this._transactionsApi
+      .getLatestTransactionsInfo(this.state.limit)
+      .then(res => {
+        this.setState({ transactions: res });
+      })
```

Comment:
I, generally, prefer to call the parameter with the name of the object field, to avoid useless naming like `res`, i.e.:

```js
.then(transactions => this.setState({ transactions }))
```

Reply:
changed